app/apps/server/bridges/livechat.js

```diff
@@ -0,0 +1,89 @@
+import { getRoom } from '../../../livechat/server/api/lib/livechat';
+import { Livechat } from '../../../livechat/server/lib/Livechat';
+import { Rooms } from '../../../models/server/models/Rooms';
+
+export class AppLivechatBridge {
+	constructor(orch) {
+		this.orch = orch;
+	}
+
+	async createMessage(message, appId) {
+		this.orch.debugLog(`The App ${ appId } is creating a new message.`);
+
+		if (!message.token) {
```

Comment:
The `token` field identifies that the message has been sent by the visitor, but we'll need to send messages when the sender is the `agent`, so I think we can't apply this verification here.

Reply:
Yes, but in the Apps-Engine side of things, if you build a `ILivechatMessage` without a properly setup visitor, it uses the message bridge instead.

https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/124/files#diff-711aff22ec7b2768c210714fac791d47R74

Would you say that's a good pattern?

Reply:
Ohh nevermind.. I didn't know about that approach.
Cool!